test/src/util.js

```diff
@@ -14,23 +15,26 @@ export function uniqueId() {
 /**
  * Register a new user with a random username.
  *
- * @returns {object} { username, email }
+ * @returns {object} { username, email, password }
  */
 export async function registerNewUser() {
   const username = `user${uniqueId()}`;
   const email = `${username}@dandi.test`;
   const password = 'password'; // Top secret
 
-  await (await page.waitForXPath(vBtn('Create Account'))).click();
+  await expect(page).toClickXPath(vBtn('Create Account'));
 
-  await (await page.waitForXPath(vTextField('Username'))).type(username);
-  await (await page.waitForXPath(vTextField('Email'))).type(email);
-  await (await page.waitForXPath(vTextField('First Name'))).type('Mister');
-  await (await page.waitForXPath(vTextField('Last Name'))).type('Roboto');
-  await (await page.waitForXPath(vTextField('Password'))).type(password);
-  await (await page.waitForXPath(vTextField('Retype password'))).type(password);
+  await expect(page).toFillXPath(vTextField('Username'), username);
```

Comment:
I thought the rule was to not use assertions / `expect` within these helpers, and instead have separate cases to directly test these functionalities?

Reply:
I personally find the `expect(page).toFillXPath(...)` syntax to be cleaner and more legible than the `await (await page.waitForXPath(...)).type(...)` syntax. In this case, it has the exact behavior I am looking for: put some words into a text field, or else throw an error if it doesn't work for some reason.
I agree it's misleading to have that `expect` statement in a helper method that isn't testing anything. When I read this it's pretty clear to me that `toFillXPath` is being used to modify the page, which I think is excusable. Requiring that all code not directly in a test method use none of the `jest-xpath` helpers sounds reasonable on paper, but obnoxious and arbitrary in practice.